From d27a64632d91717839cba23051a4c245056d16d8 Mon Sep 17 00:00:00 2001
From: balajhidn <balajhix.neelakantan.durvas@intel.com>
Date: Wed, 26 Apr 2023 09:04:19 +0000
Subject: [PATCH] Skip Firmware Update Mode Commands when SPI flash in-progress

get_bmc_fw_version is skipped  when SPI flash in-progress
This avoids access to SPI region and data corrupt

Tested:
1. Firmware Update Mode Commands query skipped when SPI flash in-progress
2. Force Firmware Update completed successfully

Signed-off-by: Gade-Rajasekhar Reddy <raja.sekhar.reddy.gade@intel.com>
Signed-off-by: balajhidn <balajhix.neelakantan.durvas@intel.com>
Upstream-Status: Pending
---
 board/aspeed/ast2600_intel/ipmi-handler.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/board/aspeed/ast2600_intel/ipmi-handler.c b/board/aspeed/ast2600_intel/ipmi-handler.c
index 3f61403ca7..8ce2b0ecb2 100644
--- a/board/aspeed/ast2600_intel/ipmi-handler.c
+++ b/board/aspeed/ast2600_intel/ipmi-handler.c
@@ -47,6 +47,8 @@
 
 #define PFM_MAGIC_NUM	0x02b3ce1d
 
+extern struct fwupd_global_setting g_fwupd_settings;
+
 typedef u16 (*fun_handler)(u8 *req, u16 req_len, u8 *res);
 
 struct get_dev_id {
@@ -171,9 +173,14 @@ static u16 get_device_id(u8 *req, u16 req_len, u8 *res)
 	/* Get Firmware version from flash(PFM Header) */
 	result->fw_rev1 = 0x00;
 	result->fw_rev2 = 0x00;
-	if(get_bmc_fw_version_from_flash(&result->fw_rev1, &result->fw_rev2)) {
-		result->completion_code = IPMI_CC_UNSPECIFIED;
-		return sizeof(result->completion_code);
+
+	// get_bmc_fw_version is skipped  when SPI flash in-progress
+	// This avoids access to SPI region and data corrupt
+	if(g_fwupd_settings.start_update == false) {
+		if(get_bmc_fw_version_from_flash(&result->fw_rev1, &result->fw_rev2)) {
+			result->completion_code = IPMI_CC_UNSPECIFIED;
+			return sizeof(result->completion_code);
+		}
 	}
 
 	/* Get Platform ID */
-- 
2.25.1

